### ISSUE TITLE
TCE 0.10 Management Cluster Creation fails "load: name crd not found in module static.lib.yaml"

### ISSUE THREAD

**Issue by appdess** [vmware-tanzu/community-edition#3189]
## Bug Report

Trying to boostrap TCE Edition with Existing AVI installation. 

Setup fails right after initiating the "tanzu management-cluster create mgmt  --file /Users/adess/.config/tanzu/tkg/clusterconfigs/adxqhx8l8h.yaml" command

```
Setting up management cluster...
Validating configuration...
Using infrastructure provider vsphere:v0.7.10
Generating cluster configuration...
WARNING: key is not a string: %!s(int=1)WARNING: key is not a string: %!s(int=1)Error: unable to set up management cluster: unable to build management cluster configuration: unable to get template:
- load: name crd not found in module static.lib.yaml
    in <toplevel>
      03_customizations/02_avi/add_akoo.yaml:5 | #@ load("static.lib.yaml", "crd","service")
```

## Expected Behavior
Boostrap of the TCE Cluster should happen and not abort

## Steps to Reproduce the Bug
Fully reproducable by running the command "tanzu management-cluster create mgmt  --file /Users/adess/.config/tanzu/tkg/clusterconfigs/adxqhx8l8h.yaml"

## Environment Details

* Build version (`tanzu version`): v0.10.1
* Deployment (Managed/Standalone cluster): Standalone
* Infrastructure Provider (Docker/AWS/Azure/vSphere): vSphere
* Operating System (client): MAC

## Diagnostics and log bundle
# Providing my config file attached below:
´´´
AVI_CA_DATA_B64: LS0tLS1CXXXXXXXXXXXXXXXXXXXXXXXX
AVI_CLOUD_NAME: Default-Cloud
AVI_CONTROL_PLANE_HA_PROVIDER: "true"
AVI_CONTROLLER: avi.haas-523.pez.vmware.com
AVI_DATA_NETWORK: Workload
AVI_DATA_NETWORK_CIDR: 10.212.189.0/24
AVI_ENABLE: "true"
AVI_LABELS: ""
AVI_MANAGEMENT_CLUSTER_VIP_NETWORK_CIDR: 10.212.189.0/24
AVI_MANAGEMENT_CLUSTER_VIP_NETWORK_NAME: Workload
AVI_PASSWORD: <encoded:LS0tLS1CXXXXXXXXXXXXXXXXXXXXXXXX>
AVI_SERVICE_ENGINE_GROUP: Default-Group
AVI_USERNAME: admin
CLUSTER_CIDR: 100.96.0.0/11
CLUSTER_NAME: mgmt
CLUSTER_PLAN: dev
ENABLE_AUDIT_LOGGING: "false"
ENABLE_CEIP_PARTICIPATION: "false"
ENABLE_MHC: "false"
IDENTITY_MANAGEMENT_TYPE: none
INFRASTRUCTURE_PROVIDER: vsphere
LDAP_BIND_DN: ""
LDAP_BIND_PASSWORD: ""
LDAP_GROUP_SEARCH_BASE_DN: ""
LDAP_GROUP_SEARCH_FILTER: ""
LDAP_GROUP_SEARCH_GROUP_ATTRIBUTE: ""
LDAP_GROUP_SEARCH_NAME_ATTRIBUTE: cn
LDAP_GROUP_SEARCH_USER_ATTRIBUTE: DN
LDAP_HOST: ""
LDAP_ROOT_CA_DATA_B64: ""
LDAP_USER_SEARCH_BASE_DN: ""
LDAP_USER_SEARCH_FILTER: ""
LDAP_USER_SEARCH_NAME_ATTRIBUTE: ""
LDAP_USER_SEARCH_USERNAME: userPrincipalName
OIDC_IDENTITY_PROVIDER_CLIENT_ID: ""
OIDC_IDENTITY_PROVIDER_CLIENT_SECRET: ""
OIDC_IDENTITY_PROVIDER_GROUPS_CLAIM: ""
OIDC_IDENTITY_PROVIDER_ISSUER_URL: ""
OIDC_IDENTITY_PROVIDER_NAME: ""
OIDC_IDENTITY_PROVIDER_SCOPES: ""
OIDC_IDENTITY_PROVIDER_USERNAME_CLAIM: ""
OS_ARCH: amd64
OS_NAME: photon
OS_VERSION: "3"
SERVICE_CIDR: 100.64.0.0/13
TKG_HTTP_PROXY_ENABLED: "false"
TKG_IP_FAMILY: ipv4
VSPHERE_CONTROL_PLANE_DISK_GIB: "20"
VSPHERE_CONTROL_PLANE_ENDPOINT: ""
VSPHERE_CONTROL_PLANE_MEM_MIB: "4096"
VSPHERE_CONTROL_PLANE_NUM_CPUS: "2"
VSPHERE_DATACENTER: /Datacenter
VSPHERE_DATASTORE: /Datacenter/datastore/Tanzu
VSPHERE_FOLDER: /Datacenter/vm/tkg
VSPHERE_NETWORK: /Datacenter/network/Tanzu/Workload
VSPHERE_PASSWORD: <encoded:R0J4SmF2dzRPN3A4TjJlZ3RpIQ==>
VSPHERE_RESOURCE_POOL: /Datacenter/host/Cluster/Resources
VSPHERE_SERVER: vcsa-01.haas-523.pez.vmware.com
VSPHERE_SSH_AUTHORIZED_KEY: ssh-rsa LS0tLS1CXXXXXXXXXXXXXXXXXXXXXXX
VSPHERE_TLS_THUMBPRINT: 4E:BD:ALS0tLS1CXXXXXXXXXXXXXXXXXXXXXXXX
VSPHERE_USERNAME: administrator@vsphere.local
VSPHERE_WORKER_DISK_GIB: "20"
VSPHERE_WORKER_MEM_MIB: "4096"
VSPHERE_WORKER_NUM_CPUS: "2"
´´´

**Comment by appdess** [comment]
Validated the same result when using a linux-machine to boostrap a mgmt-cluster with AVI (ubuntu 18.04).

```
Error: unable to set up management cluster: unable to build management cluster configuration: unable to get template:
- load: name crd not found in module static.lib.yaml
    in <toplevel>
      03_customizations/02_avi/add_akoo.yaml:5 | #@ load("static.lib.yaml", "crd","service")
```

**Comment by stmcginnis** [comment]
@seemiller do you know who would be best to take a look at this? Looks like a packaging error?

**Comment by seemiller** [comment]
Not a package problem, not sure where this would be coming from yet...

**Comment by nrobbins00** [comment]
For what it's worth, i ran into this yesterday and got past it by editing pkg/v1/providers/ytt/03_customizations/02_avi/add_akoo.yaml , removing crd from the load static.lib.yaml line so it says 
`#@ load("static.lib.yaml", "service")` 
and removing the line that says
`--- #@ crd()`

I got that from this PR https://github.com/vmware-tanzu/tanzu-framework/pull/821/files

My add_akoo.yaml looks like this now:
`
#@ load("@ytt:data", "data")
#@ load("@ytt:json", "json")

#@ if data.values.PROVIDER_TYPE == "vsphere" and data.values.TKG_CLUSTER_ROLE != "workload" and data.values.AVI_ENABLE and data.values.AVI_CONTROL_PLANE_HA_PROVIDER:
#@ load("static.lib.yaml", "service")
#@ load("rbac.lib.yaml", "role", "role_binding")
#@ load("ako-operator-deployment.lib.yaml", "akoo_deployment", "avi_credentials", "avi_certificate")

--- #@ service()
--- #@ role()
--- #@ role_binding()
--- #@ akoo_deployment()
--- #@ avi_credentials()
--- #@ avi_certificate()
#@ end
`

**Comment by nrb** [comment]
@vmware-tanzu/pkg-ako-operator-owners This looks like an issue in the AKO package, would you be able to take a look at it?

**Comment by gwang550** [comment]
I am looking into this issue now. This should be an issue that's fixed in the newer release. 

At meantime, this issue might be resolved by removing the CRD under `$HOME/.tanzu/providers` as @[nrobbins00](https://github.com/nrobbins00) commented above.

**Comment by gwang550** [comment]
I have verified the workaround works and I was able to create a management cluster after the changes. 

I have also narrowed down the problem. It seems like the current TCE is not using the latest version of the tanzu-framework. From the commit history https://github.com/vmware-tanzu/tanzu-framework/commits/main/pkg/v1/providers/ytt/03_customizations/02_avi/add_akoo.yaml, I think TCE is using the version from Oct 29, 2021 and the problem got fixed on Nov 2, 2021. This problem should be fixed once the tanzu-framework got updated.
